simulators/ethereum/rpc/eth/simulator.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+# Create and start a valid bootnode for the clients to find each other
+BOOTNODE_KEYHEX=5ce1f6fc42c817c38939498ee7fcdd9a06fa9d947d02d42dfe814406f370c4eb
+BOOTNODE_ENODEID=6433e8fb82c4638a8a6d499d40eb7d8158883219600bfd49acb968e3a37ccced04c964fa87b3a78a2da1b71dc1b90275f4d055720bb67fad4a118a56925125dc
+
+BOOTNODE_IP=`ifconfig eth0 | grep 'inet addr' | cut -d: -f2 | awk '{print $1}'`
+BOOTNODE_ENODE="enode://$BOOTNODE_ENODEID@$BOOTNODE_IP:30303"
+
+echo "Start bootnode..."
+/bootnode --nodekeyhex $BOOTNODE_KEYHEX --addr=0.0.0.0:30303 &
+
+N_NODES=5  # Number of nodes to start
+N_PEERS=4  # Number of peers for each node (N_NODES-1)
+N_MINERS=2 # Number of nodes that have the miner running
+N_LIGHT=1  # number of nodes that run in light mode
+
+# netPeerCount executes an RPC request to a node to retrieve its current number
+# of connected peers.
+#
+# If looks up the IP address of the node from the hive simulator and executes the
+# actual peer lookup. As the result is hex encoded, it will decode and return it.
+function netPeerCount {
+	local peers=`curl -sf -X POST --data '{"jsonrpc":"2.0","method":"net_peerCount","params":[],"id":0}' $1:8545 | jq '.result' | tr -d '"'`
+	echo $((16#${peers#*x}))
+}
+
+# waitPeers verifies whether a node has a given number of peers, and if not waits
+# a bit and retries a few times (to avoid devp2p pending handshakes and similar
+# anomalies). If the peer count doesn't converge to the desired target, the method
+# will fail.
+function waitPeers {
+	local ip=`curl -sf $HIVE_SIMULATOR/nodes/$1`
+
+	for i in `seq 1 20`; do
+		# Fetch the current peer count and stop if it's correct
+		peers=`netPeerCount $ip`
+		if [ "$peers" == "$2" ]; then
+		  echo "Node $1 ($ip) has $peers/$2 peers"
+			break
+		fi
+		# Seems peer count is wrong, unless too many trials, sleep a bit and retry
+		if [ "$i" == $N_NODES ]; then
```

Comment:
`$N_NODES` should be 20 here. The goal of waitPeers is to allow a few retries for reaching the desired peer count before giving up. This condition checks whether the retry count was reached or not. That count is 20 in your case per line 35.